### PR TITLE
STY: Streamline setup

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses ROADMAP, #601 

- Moves the setup metadata to setup.cfg
- Reorganized the requirements.txt and test_requirements.txt files to reduce duplication elsewhere
- Update links in community docs
- Updates travis links on readme

requirements.txt now used for pysat requirements, test_requirements.txt now used for all testing related requirements.  All testing requirements installed through pip since some of them not supported by conda.

## Type of change

- Testing fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Running `python setup.py develop`

Changes to Travis config.  Testing there.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,88 +1,12 @@
-"""A setuptools based setup module.
-See:
-https://packaging.python.org/en/latest/distributing.html
-https://github.com/pypa/sampleproject
-"""
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020, Authors
+# Full license can be found in License.md and AUTHORS.md
+# Full license can be found in License.md
```
Comment (jklenzing) [inline setup.py:5]: duplication

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,41 @@
+[metadata]
+name = pysat
+version = file: pysat/version.txt
+url = https://github.com/pysat/pysat
+author = Russell Stoneback
+author_email = rstoneba@utdallas.edu
```
Comment (jklenzing) [inline setup.cfg:6]: use pysat developers email

File: setup.cfg [hunk 3]
```diff
@@ -1,3 +1,41 @@
...
+  Topic :: Scientific/Engineering :: Atmospheric Science
+  License :: OSI Approved :: BSD License
+  Natural Language :: English
+  Programming Language :: Python :: 3.6
+  Programming Language :: Python :: 3.7
+  Programming Language :: Python :: 3.8
+  Operating System :: MacOS :: MacOS X
+  Operating System :: POSIX :: Linux
+license_file = LICENSE
+long_description = file: description.txt
```
Comment (jklenzing) [inline setup.cfg:25]: add an issue to ultimately use a scrubbed version of readme.md here

Comment (aburrell) [reply setup.cfg:25]: Added this to #607

**Review by rstoneback — APPROVED** [review]
Thanks @jklenzing!

**Pull request merged**